controllers/reconcile.go

```diff
@@ -4,67 +4,117 @@ import (
-		return err
+	if err := r.reconcileRolloutsAggregateToAdminClusterRole(ctx); err != nil {
+		log.Error(err, "failed to reconcile Rollout's aggregate-to-admin ClusterRoles.")
+		return createCondition(err.Error()), err
 	}
 
 	log.Info("reconciling aggregate-to-edit ClusterRole")
-	if err := r.reconcileRolloutsAggregateToEditClusterRole(ctx, cr); err != nil {
-		return err
+	if err := r.reconcileRolloutsAggregateToEditClusterRole(ctx); err != nil {
+		log.Error(err, "failed to reconcile Rollout's aggregate-to-edit ClusterRoles.")
+		return createCondition(err.Error()), err
 	}
 
 	log.Info("reconciling aggregate-to-view ClusterRole")
-	if err := r.reconcileRolloutsAggregateToViewClusterRole(ctx, cr); err != nil {
-		return err
+	if err := r.reconcileRolloutsAggregateToViewClusterRole(ctx); err != nil {
+		log.Error(err, "failed to reconcile Rollout's aggregate-to-view ClusterRoles.")
+		return createCondition(err.Error()), err
 	}
 
-	log.Info("reconciling rollouts role bindings")
-	if err := r.reconcileRolloutsRoleBinding(ctx, cr, role, sa); err != nil {
-		return err
+	if cr.Spec.NamespaceScoped {
+		log.Info("reconciling Rollouts RoleBindings")
+		if err := r.reconcileRolloutsRoleBinding(ctx, cr, role, sa); err != nil {
+			log.Error(err, "failed to reconcile Rollout's RoleBindings.")
+			return createCondition(err.Error()), err
+		}
+	} else {
+		log.Info("reconciling Rollouts ClusterRoleBinding")
+		if err := r.reconcileRolloutsClusterRoleBinding(ctx, clusterRole, sa); err != nil {
+			log.Error(err, "failed to reconcile Rollout's ClusterRoleBinding.")
+			return createCondition(err.Error()), err
+		}
 	}
 
-	log.Info("reconciling rollouts secret")
+	log.Info("reconciling Rollouts Secret")
 	if err := r.reconcileRolloutsSecrets(ctx, cr); err != nil {
-		return err
+		log.Error(err, "failed to reconcile Rollout's Secret.")
+		return createCondition(err.Error()), err
 	}
 
-	// reconcile configMap for plugins
-	log.Info("reconciling configMap for plugins")
+	log.Info("reconciling ConfigMap for plugins")
 	if err := r.reconcileConfigMap(ctx, cr); err != nil {
-		return err
+		log.Error(err, "failed to reconcile Rollout's ConfigMap.")
+		return createCondition(err.Error()), err
 	}
 
-	log.Info("reconciling rollouts deployment")
+	log.Info("reconciling Rollouts Deployment")
 	if err := r.reconcileRolloutsDeployment(ctx, cr, *sa); err != nil {
-		return err
+		log.Error(err, "failed to reconcile Rollout's Deployment.")
+		return createCondition(err.Error()), err
 	}
 
-	log.Info("reconciling rollouts metrics service")
+	log.Info("reconciling Rollouts Metrics Service")
 	if err := r.reconcileRolloutsMetricsService(ctx, cr); err != nil {
-		return err
+		log.Error(err, "failed to reconcile Rollout's Metrics Service.")
+		return createCondition(err.Error()), err
 	}
 
 	log.Info("reconciling status of workloads")
 	if err := r.reconcileStatus(ctx, cr); err != nil {
-		return err
+		log.Error(err, "failed to reconcile Rollout's Metrics Service.")
```

Comment:
`failed to reconcile Rollout's Metrics Service.` -> `failed to reconcile status of workloads`